admin/client/src/containers/FormBuilderLoader/FormBuilderLoader.js

```diff
@@ -161,27 +192,34 @@ FormBuilderLoader.propTypes = Object.assign({}, basePropTypes, {
   form: PropTypes.string,
   submitting: PropTypes.bool,
 });
+
 FormBuilderLoader.defaultProps = {
   // Perform this *outside* of render() to avoid re-rendering of the whole DOM structure
   // every time render() is triggered.
   baseFormComponent: reduxForm()(Form),
   baseFieldComponent: ReduxFormField,
 };
 
-export default connect(
-  (state, ownProps) => {
-    const schema = state.schemas[ownProps.schemaUrl];
-    const form = schema && schema.id;
-    const reduxFormState = state.form
-      && state.form[ownProps.schemaUrl];
-    const submitting = reduxFormState
-      && reduxFormState.submitting;
-    const values = reduxFormState
-      && reduxFormState.values;
-
-    return { schema, form, submitting, values };
-  },
-  (dispatch) => ({
+function mapStateToProps(state, ownProps) {
+  const schema = state.schemas[ownProps.schemaUrl];
+  const form = schema && schema.id;
+
+  const reduxFormState = state.form && state.form[ownProps.schemaUrl];
+  const submitting = reduxFormState && reduxFormState.submitting;
+  const values = reduxFormState && reduxFormState.values;
+
+  const formData = state.formData[ownProps.schemaUrl];
+  const stateOverrides = formData && formData.stateOverride;
+  const loading = formData && formData.loading;
```

Comment:
Why is the `loading` flag on `state.formData` rather than `state.schemas`? It seems to make it harder to iterate through actual form data, because there's metadata mixed in.